src/telemetry/Manager.cc

```diff
@@ -38,6 +43,232 @@ void Manager::InitPostBrokerSetup(broker::endpoint& ep)
+	zeek_int_t metric_type_int = -1;
+
+	// This is mapping internal enum values for the types to the
+	// right EnumVal. TODO: There might be a better way.
+	switch ( metric_type_ )
+		{
+		case DOUBLE_COUNTER:
+			metric_type_int = et->Lookup("Telemetry::DOUBLE_COUNTER");
+			break;
+		case INT_COUNTER:
+			metric_type_int = et->Lookup("Telemetry::INT_COUNTER");
+			break;
+		case DOUBLE_GAUGE:
+			metric_type_int = et->Lookup("Telemetry::DOUBLE_GAUGE");
+			break;
+		case INT_GAUGE:
+			metric_type_int = et->Lookup("Telemetry::INT_GAUGE");
+			break;
+		case DOUBLE_HISTOGRAM:
+			metric_type_int = et->Lookup("Telemetry::DOUBLE_HISTOGRAM");
+			break;
+		case INT_HISTOGRAM:
+			metric_type_int = et->Lookup("Telemetry::INT_HISTOGRAM");
+			break;
+		}
+
+	if ( metric_type_int < 0 )
+		reporter->FatalError("Unable to lookup %d", int(metric_type_));
+
+	zeek::EnumValPtr metric_type_val = et->GetEnumVal(metric_type_int);
+	r->Assign(rt_off_metric_type, metric_type_val);
+
+	// TODO: Handle bounds for histograms
+	// if (metric_type_ == DOUBLE_HISTOGRAM || metric_type == INT_HISTOGRAM)
+
+	return r;
+	}
+
+zeek::RecordValPtr Manager::Metric::get_metric_record()
+	{
+	static auto svt = zeek::id::find_type<zeek::VectorType>("string_vec");
+	static auto rt = zeek::id::find_type<zeek::RecordType>("Telemetry::Metric");
+	static auto rt_off_opts = rt->FieldOffset("opts");
+	static auto rt_off_label_values = rt->FieldOffset("label_values");
+	static auto rt_off_value = rt->FieldOffset("value");
+	static auto rt_off_count_value = rt->FieldOffset("count_value");
+
+	auto r = make_intrusive<zeek::RecordVal>(rt);
+	// TODO: Cache this somewhere. Maybe do not set it here at all.
+	auto opts_record = get_metric_opts_record();
+	r->Assign(rt_off_opts, opts_record);
+
+	if ( value_ )
+		r->Assign(rt_off_value, make_intrusive<zeek::DoubleVal>(*value_));
+	if ( count_value_ )
+		r->Assign(rt_off_count_value, val_mgr->Count(*count_value_));
+
+	auto label_values_vec = make_intrusive<zeek::VectorVal>(svt);
+	int i = 0;
+	for ( auto const& l : labels_ )
+		label_values_vec->Assign(i++, make_intrusive<StringVal>(l));
+	r->Assign(rt_off_label_values, label_values_vec);
+
+	return r;
+	}
+
+namespace bt = broker::telemetry;
+
+/**
+ * A collector implementation fetching info and values (for counters and guages);
+ */
+class MetricsCollector : public broker::telemetry::metrics_collector
+	{
+	using MT = Manager::Metric::MetricType;
+
+public:
+	MetricsCollector(std::string_view prefix, std::string_view name)
+		: prefix_pattern_(prefix), name_pattern_(name)
+		{
+		}
+
+	void operator()(const bt::dbl_counter_family_hdl* family, const bt::dbl_counter_hdl* counter,
+	                const std::vector<std::string_view>& label_values)
+		{
+		if ( matches(upcast(family)) )
+			metrics_.push_back({MT::DOUBLE_COUNTER, bt::upcast(family), (double)bt::value(counter),
+			                    std::nullopt, label_values});
```

Comment:
Style nit, I  *believe* this would work too:

```suggestion
			metrics_.emplace_back(MT::DOUBLE_COUNTER, bt::upcast(family), (double)bt::value(counter),
			                    std::nullopt, label_values);
```